modules/auxiliary/server/capture/ldap.rb

```diff
@@ -0,0 +1,181 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Auxiliary::Report
+  include Msf::Exploit::Remote::LDAP::Server
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Authentication Capture: LDAP',
+        'Description' => %q{
+          This module mocks an LDAP service to capture authentication
+          information of a client trying to authenticate against an LDAP service
+        },
+        'Author' => 'JustAnda7',
+        'License' => MSF_LICENSE,
+        'Action' => [
+          [ 'Capture', { 'Description' => 'Run an LDAP capture server' } ]
+        ],
+        'PassiveActions' => [ 'Capture' ],
+        'DefaultActions' => 'Capture',
+        'Notes' => {
+          'Stability' => [],
+          'Reliability' => [],
+          'SideEffects' => []
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptAddress.new('SRVHOST', [ true, 'The localhost to listen on.', '0.0.0.0' ]),
+        OptPort.new('SRVPORT', [ true, 'The local port to listen on.', '389' ]),
+        OptString.new('Authentication', [ false, 'The type of authentication used by the client.', 'Simple' ])
+      ]
+    )
+  end
+
+  def setup
+    super
+    @state = {}
```

Comment:
Why is this an instance variable?

Looking at usage it could may be viable to scope this as a private variable in `on_dispatch_request`.

Reply:
I will move it in `on_dispatch_request`. We can move it out when required